### PR TITLE
net-news/sfeed: add v1.3 and some improvements

### PR DESCRIPTION
**Pull request by N-R-K**
removes unneeded depandencies
adds theme support via USE flag and savedconfig
downgrade to EAPI 7 (8 doesn't support savedconfig)

Closes: https://bugs.gentoo.org/834240

### CODE REVIEW

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 1]
```diff
@@ -45,4 +49,6 @@ src_install() {
 		DOCPREFIX="${EPREFIX}/usr/share/doc/${P}" \
```
Comment (AndrewAmmerlaan) [inline net-news/sfeed/sfeed-1.2.ebuild:49]:
```suggestion
		DOCPREFIX="${EPREFIX}/usr/share/doc/${PF}" \
```
Correct location for the documentation is `PF` not `P` (PF also includes the revision)

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 2]
```diff
@@ -12,29 +12,33 @@ SRC_URI="https://codemadness.org/releases/${PN}/${P}.tar.gz"
...
+
+	restore_config $(printf "themes/%s.h " ${THEMES[@]//-/_})
+}
 
 src_compile() {
 	emake \
 		CC=$(tc-getCC) \
 		RANLIB=$(tc-getRANLIB) \
 		AR=$(tc-getAR) \
-		SFEED_CURSES=$(usex ncurses "sfeed_curses" "")
```
Comment (ionenwks) [inline net-news/sfeed/sfeed-1.2.ebuild:37]: Unrelated to the changes here, but the CC/RANLIB/AR above this line are lacking quotes

if for example had CC="gcc -m32" (or anything else, but -m32 is a common case) it'll fail to build.

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 3]
```diff
@@ -12,29 +12,33 @@ SRC_URI="https://codemadness.org/releases/${PN}/${P}.tar.gz"
 LICENSE="ISC"
 SLOT="0"
 KEYWORDS="~amd64"
-IUSE="ncurses"
-
-DEPEND="ncurses? ( sys-libs/ncurses )"
-RDEPEND="${DEPEND}
-	net-misc/curl
```
Comment (ionenwks) [inline net-news/sfeed/sfeed-1.2.ebuild:19]: curl is used by sfeed_update and shouldn't be considered an implicit system dep (unlike coreutils, glibc, and virtual/awk to a rough extend).

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 4]
```diff
@@ -12,29 +12,33 @@ SRC_URI="https://codemadness.org/releases/${PN}/${P}.tar.gz"
...
-	sys-apps/coreutils
-	sys-libs/glibc
-	virtual/awk
-	www-client/lynx
-	x11-misc/xclip
-	x11-misc/xdg-utils"
```
Comment (ionenwks) [inline net-news/sfeed/sfeed-1.2.ebuild:25]: fwiw these are used by sfeed_curses/_content, lynx notably to dump html to text (although could be semi-optfeatures, not that I've tried sfeed I don't have more to say).

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 5]
```diff
@@ -12,29 +12,33 @@ SRC_URI="https://codemadness.org/releases/${PN}/${P}.tar.gz"
...
+
+DEPEND="ncurses? ( sys-libs/ncurses:= )"
+RDEPEND="${DEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+src_configure() {
+	for name in "${THEMES[@]}"; do
+		if use theme-${name}; then
+			SFEED_THEME="${name//-/_}"
+		fi
```
Comment (ionenwks) [inline net-news/sfeed/sfeed-1.2.ebuild:28]: just style but I'd `use theme-${name} &&`

**Review by ionenwks — COMMENTED** [review]
Just a quick overview given I remembered the pkg-config bothering me when I seen the guru ebuild.

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 6]
```diff
@@ -12,37 +12,43 @@ SRC_URI="https://codemadness.org/releases/${PN}/${P}.tar.gz"
...
+THEMES=( mono{,-highlight} newsboat templeos )
+IUSE="+ncurses +$(printf "theme-%s " ${THEMES[@]})"
+REQUIRED_USE="ncurses? ( ^^ ( $(printf "theme-%s " ${THEMES[@]}) ) )"
+
+DEPEND="ncurses? ( sys-libs/ncurses:= )"
+RDEPEND="${DEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+src_configure() {
+	for name in "${THEMES[@]}"; do
```
Comment (ionenwks) [inline net-news/sfeed/sfeed-1.2.ebuild:25]: One more thing, missing `local name`

File: net-news/sfeed/sfeed-1.2.ebuild (hunk 3 above)
Comment (N-R-K) [reply net-news/sfeed/sfeed-1.2.ebuild:19]: curl is an optional runtime dep. should be opt-feature i think.

File: net-news/sfeed/sfeed-1.2.ebuild (hunk 4 above)
Comment (N-R-K) [reply net-news/sfeed/sfeed-1.2.ebuild:25]: same as curl, these are all optional runtime deps. I'll put them into opt-feature.

**Comment by N-R-K** [comment]
> Just a quick overview given I remembered the pkg-config bothering me when I seen the guru ebuild.

Thanks for the review. I'm off for today though, will fix the issues tomorrow.

File: net-news/sfeed/sfeed-1.2.ebuild [hunk 7]
```diff
@@ -12,29 +12,33 @@ SRC_URI="https://codemadness.org/releases/${PN}/${P}.tar.gz"
...
 
 src_compile() {
 	emake \
 		CC=$(tc-getCC) \
 		RANLIB=$(tc-getRANLIB) \
 		AR=$(tc-getAR) \
-		SFEED_CURSES=$(usex ncurses "sfeed_curses" "")
+		SFEED_CURSES=$(usex ncurses "sfeed_curses" "") \
+		SFEED_THEME="${SFEED_THEME}" \
+		SFEED_CURSES_LDFLAGS="${LDFLAGS} $(pkg-config --libs ncurses)"
```
[the remark this answers is not in the document]
Comment (N-R-K) [reply net-news/sfeed/sfeed-1.2.ebuild:41]: i'll fix it tomorrow, thanks.

_NOTE to self: sfeed_curses offers a bundled minicurses library as alternative to ncurses. should probably add a useflag for that..._

**Comment by N-R-K** [comment]
New version of sfeed was released just today. Mostly was documentation related change.

Any suggestion on what to do in this situation? I'm thinking about leaving the current ebuild as is and adding the new version (with the improvements) instead.

**Pull request closed without merging**